apihub-portal-publish/deploy-unmanaged-api.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if [ -z "$PROJECT_ID" ]; then
+  echo "No PROJECT_ID variable set"
+  exit
+fi
+
+if [ -z "$REGION" ]; then
+  echo "No REGION variable set"
+  exit
+fi
+
+if [ -z "$APIGEE_ENV" ]; then
+  echo "No APIGEE_ENV variable set"
+  exit
+fi
+
+if [ -z "$APIGEE_HOST" ]; then
+  echo "No APIGEE_HOST variable set"
+  exit
+fi
+
+if [ -z "$APIGEE_PORTAL_URL" ]; then
+  echo "No APIGEE_PORTAL_URL variable set"
+  exit
+fi
+
+echo "💻 Registering an unmanaged API in API Hub to the 'Test' environment..."
+
+# copy definitions and set env variables in local files
+cp apihub-api.json apihub-api.local.json
+sed -i "s,PROJECT_ID,$PROJECT_ID,g" ./apihub-api.local.json
+sed -i "s,REGION,$REGION,g" ./apihub-api.local.json
+cp apihub-api-version.json apihub-api-version.local.json
+sed -i "s,PROJECT_ID,$PROJECT_ID,g" ./apihub-api-version.local.json
+sed -i "s,REGION,$REGION,g" ./apihub-api-version.local.json
+cp apihub-api-deployment.json apihub-api-deployment.local.json
+sed -i "s,PROJECT_ID,$PROJECT_ID,g" ./apihub-api-deployment.local.json
+sed -i "s,REGION,$REGION,g" ./apihub-api-deployment.local.json
+
+# create deployment
+curl -X POST "https://apihub.googleapis.com/v1/projects/$PROJECT_ID/locations/$REGION/deployments?deploymentId=apigee-sample-unmanaged-v1-deployment" \
```

Comment:
why not use apigeecli for this?

Reply:
I tried to use apigeecli for the deployment like for everything else, but couldn't find a way to apply a JSON, since I need to set more properties than are offered in the CLI helper function, see doc [here](https://github.com/apigee/apigeecli/blob/main/docs/apigeecli_apihub_deployments_create.md).

In case it's possible to pass a file like for [version](https://github.com/apigee/apigeecli/blob/main/docs/apigeecli_apihub_apis_versions_create.md) then happy to use it..